server/api/crud/alerts.py

```diff
@@ -191,6 +191,14 @@ def process_event(
                 event=event_data.entity.ids[0],
             )
 
+    @staticmethod
+    def populate_event_cache(session: sqlalchemy.orm.Session):
+        for alert in server.api.utils.singletons.db.get_db().get_all_alerts(session):
+            for event_name in alert.trigger.events:
+                server.api.crud.Events().add_event_configuration(
+                    alert.project, event_name, alert.name
+                )
```

Comment:
this will result in N*M access to DB where N = alerts, M = events.
I suggest to revise the query and make it more efficient wrt accessing DB